src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -398,7 +398,20 @@ public IEnumerable<CustomNodeInfo> AddUninitializedCustomNodesInPath(string path
             {
                 info.IsPackageMember = isPackageMember;
 
-                SetNodeInfo(info, isTestMode);
+                SetNodeInfo(info);
+                result.Add(info);
+            }
+            return result;
+        }
+
+        public IEnumerable<CustomNodeInfo> AddUninitializedCustomNodesInPath(string path, bool isTestMode, PackageInfo packageInfo)
+        {
+            var result = new List<CustomNodeInfo>();
+            foreach (var info in ScanNodeHeadersInDirectory(path, isTestMode))
```

Comment:
Without checking the function `ScanNodeHeadersInDirectory` can you tell us a bit why testMode is needed here?

Reply:
so because this function results in loading something from disk, its needed to be passed down to the xml and json deserialization methods:

https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Core/CustomNodeManager.cs#L601

eventually... it's used to fail tests, which is a good thing (instead of logging the exceptions like we do at runtime)

https://github.com/DynamoDS/Dynamo/blob/ec10f936824152e7dd7d6d019efdcda0d78a5264/src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs#L245